tests/test_game_wrapper_pokemon_pinball.py

```diff
@@ -0,0 +1,115 @@
+    assert pyboy.cartridge_title == "POKEPINBALLVPH"
+
+    pokemon_pinball = pyboy.game_wrapper
+    pokemon_pinball.start_game()
+
+    assert pokemon_pinball.score == 0
+    assert pokemon_pinball.balls_left == 2
+    assert pokemon_pinball.current_stage == Stage.RED_BOTTOM.value
+    assert pokemon_pinball.special_mode_active == False
+    pyboy.stop()
+
+
+def test_pokemon_pinball_advanced(pokemon_pinball_rom):
+    pyboy = PyBoy(pokemon_pinball_rom, game_wrapper=True, window_type="null")
+    pyboy.set_emulation_speed(0)
+    assert pyboy.cartridge_title == "POKEPINBALLVPH"
+
+    pokemon_pinball = pyboy.game_wrapper
+    pokemon_pinball.start_game(stage=Stage.BLUE_BOTTOM)
+    pyboy.button_press("a")
+    pyboy.button_press("left")
+    for i in range(500):
+        pyboy.tick(render=False)
+    pyboy.button_release("left")
+    pyboy.button_release("a")
+    for i in range(20):
+        pyboy.tick(render=False)
+    pyboy.button_press("left")
+    pyboy.button_press("a")
+    for i in range(700):
+        pyboy.tick(render=False)
+
+    assert pokemon_pinball.score == 123200
+    assert pokemon_pinball.special_mode == SpecialMode.CATCH.value
+    assert pokemon_pinball.current_stage == Stage.BLUE_BOTTOM.value
+    assert pokemon_pinball.special_mode_active == True
+    assert pokemon_pinball.balls_left == 2
+
+    pyboy.stop()
+
+
+def test_pokemon_catch_mode(pokemon_pinball_rom):
+    pyboy = PyBoy(pokemon_pinball_rom, game_wrapper=True, window_type="null")
+    pyboy.set_emulation_speed(0)
+    assert pyboy.cartridge_title == "POKEPINBALLVPH"
+
+    pokemon_pinball = pyboy.game_wrapper
+    pokemon_pinball.start_game(stage=Stage.RED_BOTTOM)
+    pyboy.button_press("a")
+    pyboy.button_press("left")
+    for i in range(50):
+        pyboy.tick(render=False)
+    pokemon_pinball.start_catch_mode()
+    for i in range(270):
+        pyboy.tick(render=False)
+    pyboy.button_release("left")
+    pyboy.button_release("a")
+    pyboy.button("select")
+    for i in range(20):
+        pyboy.tick(render=False)
+    pyboy.button_press("left")
+    pyboy.button_press("a")
+    for i in range(500):
+        pyboy.tick(render=False)
+    pyboy.button_release("left")
+    for i in range(21):
+        pyboy.tick(render=False)
+    pyboy.button_press("left")
+    for i in range(700):
+        pyboy.tick(render=False)
+
+    assert pokemon_pinball.score == 15635100
+    assert pokemon_pinball.has_pokemon(Pokemon.BULBASAUR)
+    assert pokemon_pinball.has_pokemon(Pokemon.CHARMANDER) == False
+    assert pokemon_pinball.get_unique_pokemon_caught() == 1
+
+    pyboy.stop()
+
+
+def test_pokemon_pinball_game_over(pokemon_pinball_rom):
+    pyboy = PyBoy(pokemon_pinball_rom, game_wrapper=True, window_type="null")
+    pyboy.set_emulation_speed(0)
+    assert pyboy.cartridge_title == "POKEPINBALLVPH"
+
+    pokemon_pinball = pyboy.game_wrapper
+    pokemon_pinball.start_game()
+    pyboy.button("a")
+    for i in range(6185):
```

Comment:
This is a very specific number. How was it derived?

Reply:
just did some further testing, the actual value that should go there is 6176. this will be the first tick that game over is true if you press a every 100 ticks, which essentially just launches the ball as soon as its up, and doesnt really do anything to prevent it from going into the loss slot repeatedly.

Originally it was just trial and error by hand, started with a big number i knew would be enough for a game over and started lowering until it was just before the game over occurred.